test/fixtures/linter/autofix/ansi/016_no_fix_in_template_loops/after.sql

```diff
@@ -0,0 +1,7 @@
+SELECT
+    {% for _ in [1, 2, 3] %}      {% endfor %}
```

Comment:
Why have we lost the newline from before?

Reply:
I honestly don't know. Think we should address it in this PR or create a new one? 

I think we've gotten into an area of SQLFluff with tons of squirrelly behavior. 😛 I can tell this because of the number of "no cover" comments (I'm seeing now how these can be helpful!). I worked on a somewhat similar project in the early 2000s, where there was a 3-6 month period that, whenever you fixed a bug, you'd find two more. It felt pretty crazy at the time, but we kept adding new test cases as we found them. Eventually, it became really stable and Oracle bought us, but I was gone by then. 

https://www.oracle.com/industries/retail/products/science-platform/

Reply:
> Think we should address it in this PR or create a new one?

Not sure. But makes me nervous that we're proposing a heuristic-based fix, that I don't fully understand if I'm honest (no offence to your code - it's just complicated area as you say), and even the test cases are showing "funnies"...

Reply:
>Why have we lost the newline from before?

This is no longer happening.

Reply:
Looks better in latest code. Resolving.